restygwt/src/main/java/org/fusesource/restygwt/rebind/RestServiceClassCreator.java

```diff
@@ -936,7 +936,11 @@ protected String toStringExpression(JType type, String expr) throws UnableToComp
 
         if (DATE_TYPE == type) {
             String encoded = locator.encodeExpression(DATE_TYPE, expr, Style.DEFAULT);
-            return "java.util.Objects.toString(" + encoded + ", null)";
+            return String.format(
+                    "%s instanceof com.google.gwt.json.client.JSONString ?" +
+                            " ((com.google.gwt.json.client.JSONString) %s).stringValue() :" +
+                            " java.util.Objects.toString(%s, null)",
```

Comment:
I never saw this Objects class so I look up the javadocs and it turns out that `java.util.Objects.toString(%s)` would already do the right thing ;)

Reply:
Nop it's not the same:
`Objects.toString(null)` returns the String`"null"`
`Objects.toString(null, null)` returns `null`

Reply:
oh. cool